### PR TITLE
Update settings_backup.php

### PR DESCRIPTION
**Pull request by aftechro**
Backing up the sql and server document root as a tar.gz archive To do: Backup scheduler, restore from backup

### CODE REVIEW

**Comment by wrongecho** [comment]
Test these changes at: https://patch14864.pr-review.itflow.org  
 _(automatic message)_

**Comment by wrongecho** [comment]
Test these changes at: https://patch14864.pr-review.itflow.org  
 _(automatic message)_

File: settings_backup.php [hunk 1]
```diff
@@ -1,37 +1,399 @@
...
+$conn = new mysqli($dbhost, $dbusername, $dbpassword, $database);
+
+// Get the server's document root
+$documentRootPath = $_SERVER['DOCUMENT_ROOT'];
+
+// Handle backup action
+if ($_SERVER["REQUEST_METHOD"] === "POST" && isset($_POST['backup'])) {
+    // Create a SQL backup excluding comments
+    $sqlBackupFile = $documentRootPath . '/db_backup.sql';
+    $sqlDumpCommand = "mysqldump --complete-insert --skip-comments --skip-triggers --host=$dbhost --user=$dbusername --password=$dbpassword $database > $sqlBackupFile";
```
[the remark this answers is not in the document]
Comment (aftechro) [reply settings_backup.php:27]: false positive

**Comment by wrongecho** [comment]
Closing as this was just used for testing via PR Review :)

**Comment by aftechro** [comment]
> Closing as this was just used for testing via PR Review :)

yesterday i was looking at it haha

**Pull request closed without merging**